modules/apps/object/object-test/src/testFunctional/tests/ObjectFieldDefaultValue.testcase

```diff
@@ -45,12 +45,130 @@ definition {
+				fieldName = "picklistField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject181204",
+				picklistName = "Custom Picklist 181204");
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject181204");	
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject181204");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 181204");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.goToFieldsDetails(label = "Picklist Field");
+
+			Click(locator1 = "ObjectAdmin#SIDEBAR_ADVANCED_TAB");
+
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Use Default Value",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Click(
+				key_optionDefaulValue = "Expression Builder",
+				locator1 = "ObjectAdmin#PICKLIST_INPUT_AS_VALUE_OR_EXPRESSION");
+
+			Type(
+				key_text = "Default Value",
+				locator1 = "TextInput#ANY",
+				value1 = "customObject");
+
+			Button.clickSave();
+		}
+
+		task ("And given an object entry with no item selected") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject181204");
+
+			ObjectAdmin.addObjectStringFieldEntryViaUI(entryValue = "Entry Test");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject181204");
 
+			ObjectPortlet.viewEntry(entry = "Entry Test");
+
+			AssertElementNotPresent(
+				key_entry = "Item Test 1",
+				locator1 = "ObjectPortlet#ENTRY_VALUE");
+		}
+
+		task ("When the default value configuration is updated to Input as Value") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object 181204");
+
+			ObjectAdmin.goToFieldsTab();
+
+			ObjectAdmin.goToFieldsDetails(label = "Picklist Field");
+
+			Click(locator1 = "ObjectAdmin#SIDEBAR_ADVANCED_TAB");
+
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Use Default Value",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Click(
+				key_optionDefaulValue = "Input as Value",
+				locator1 = "ObjectAdmin#PICKLIST_INPUT_AS_VALUE_OR_EXPRESSION");
+
+			Click(locator1 = "ObjectAdmin#DROPDOWN_DEFAULT_VALUE");
+
+			Click(
+				key_dropdownItem = "Item Test 1",
+				locator1 = "ObjectAdmin#SELECT_DROPDOWN_ITEM_DEFAULT_VALUE");
+
+			Button.clickSave();
+		}
+
+		task ("Then the object entry is updated and the picklist field is set to the default value") {
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject181204");
+
+			ObjectPortlet.viewEntry(entry = "Entry Test");
+
+			AssertElementNotPresent(
+				key_entry = "Item Test 1",
+				locator1 = "ObjectPortlet#ENTRY_VALUE");
+
+			ObjectAdmin.addObjectStringFieldEntryViaUI(entryValue = "New Entry Test");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject181204");
+
+			ObjectPortlet.viewEntry(entry = "Item Test 1");
```

Comment:
@andreefarias can you weigh in here? I thought the expected behavior was for the existing entries to be updated to use the default value. Is that wrong? If so, we need to modify this test.

Reply:
No no. The expected behavior is only valid when creating entries, not updating.

cc: @timpak

Reply:
I reviewed the BDD steps and test name, I guess that makes more sense now. What do you think?

Ticket: [LPS-181204](https://issues.liferay.com/browse/LPS-181204)

OBS: The Expression Builder was removed for this user story.